rpcadaptors.go

```diff
@@ -154,21 +215,54 @@ func (cm *rpcConnManager) RemoveByID(id int32) error {
 // This function is safe for concurrent access and is part of the
 // rpcserver.ConnManager interface implementation.
 func (cm *rpcConnManager) RemoveByAddr(addr string) error {
-	replyChan := make(chan error)
-	cm.server.query <- removeNodeMsg{
-		cmp:   func(sp *serverPeer) bool { return sp.Addr() == addr },
-		reply: replyChan,
+	cmp := func(sp *serverPeer) bool { return sp.Addr() == addr }
+	err := cm.removeNode(cmp)
+	if err != nil {
+		netAddr, err := addrStringToNetAddr(addr)
+		if err != nil {
+			return err
+		}
+		return cm.server.connManager.CancelPending(netAddr)
```

Comment:
Good observation.

After looking into it a bit, it's because `CancelPending` needs an address which is not available in the ID path.  While technically, it would need to be called there too, a bit of an overhaul would be needed to make it possible.  Also, in practice, since peers don't show up in `getpeerinfo` results until they've successfully negotiated, there isn't a remote way for a caller to know the ID of a pending peer anyway.

That said, the focus of this PR is on separating the state without changing any of the other semantics, so this would need to be addressed in a separate PR.